CI/deploy-mudlet-for-windows.sh

```diff
@@ -314,27 +319,33 @@ else
   if [[ "$PublicTestBuild" == "true" ]]; then
     echo "=== Downloading release feed ==="
     DownloadedFeed=$(mktemp)
-    curl "https://feeds.dblsqd.com/MKMMR7HNSP65PquQQbiDIw/public-test-build/win/x86" -o "$DownloadedFeed"
+    curl "https://feeds.dblsqd.com/MKMMR7HNSP65PquQQbiDIw/public-test-build/win/${DBLSQDTYPE}" -o "$DownloadedFeed"
     
     echo "=== Generating a changelog ==="
     cd "$GITHUB_WORKSPACE/CI" || exit 1
-    Changelog=$(lua "${GITHUB_WORKSPACE}/CI/generate-changelog.lua" --mode ptb --releasefile "$DownloadedFeed")
+    Changelog=$(lua5.1 "${GITHUB_WORKSPACE}/CI/generate-changelog.lua" --mode ptb --releasefile "$DownloadedFeed")
     cd - || exit 1
     echo "$Changelog"
     
     echo "=== Creating release in Dblsqd ==="
-    dblsqd release -a mudlet -c public-test-build -m "$Changelog" "${VERSION}${MUDLET_VERSION_BUILD}-${BUILD_COMMIT,,}"
-
-    echo "=== Registering release with Dblsqd ==="
-    dblsqd push -a mudlet -c public-test-build -r "${VERSION}${MUDLET_VERSION_BUILD}-${BUILD_COMMIT,,}" -s mudlet --type 'standalone' --attach "win:${DBLSQDTYPE}" "${DEPLOY_URL}"
+    VersionString="${VERSION}${MUDLET_VERSION_BUILD}-${BUILD_COMMIT,,}"
+    export VersionString
 
+    # This may fail as a build from another architecture may have already registered a release with dblsqd,
+    # if so, that is OK...
+    echo "=== Creating release in Dblsqd ==="
+    echo "dblsqd release -a mudlet -c public-test-build -m \"$Changelog\" \"${VersionString}\""
+    dblsqd release -a mudlet -c public-test-build -m "$Changelog" "${VersionString}" || true
   fi
 fi
 
 if [[ -n "$GITHUB_PULL_REQUEST_NUMBER" ]]; then
   prId=" ,#$GITHUB_PULL_REQUEST_NUMBER"
 fi
 
+# Make PublicTestBuild available GHA to check if we need to run the register step
+echo "PUBLIC_TEST_BUILD=${PublicTestBuild}" >> "$GITHUB_ENV"
```

Comment:
I was considering this.. The release build is uploaded to the wp-downloadmanager instead of as an artifact, I need to make sure this looks correct:
`echo "=== Uploading installer to https://www.mudlet.org/wp-content/files/?C=M;O=D ==="
    scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null "$installerExePath" "mudmachine@mudlet.org:${DEPLOY_PATH}"
    DEPLOY_URL="https://www.mudlet.org/wp-content/files/Mudlet-${VERSION}-windows-$BUILD_BITNESS-installer.exe"

    SHA256SUM=$(shasum -a 256 "$installerExePath" | awk '{print $1}')

    # file_cat=0 assuming Windows is the 0th item in WP-Download-Manager category
    curl -X POST 'https://www.mudlet.org/wp-content/plugins/wp-downloadmanager/download-add.php' \
    -H "x-wp-download-token: $X_WP_DOWNLOAD_TOKEN" \
    -F "file_type=2" \
    -F "file_remote=$DEPLOY_URL" \
    -F "file_name=Mudlet-${VERSION} (windows-$BUILD_BITNESS)" \
    -F "file_des=sha256: $SHA256SUM" \
    -F "file_cat=0" \
    -F "file_permission=-1" \
    -F "output=json" \
    -F "do=Add File"`
    
I'm not sure where DEPLOY_PATH is being set, it isn't being set anywhere that I could find in any of the old build scripts for any platform.
And some clarification on how the download manager works..
We have the installerExePath being set here:
`installerExePath="Mudlet-$VERSION$MUDLET_VERSION_BUILD-$BUILD_COMMIT-windows-$BUILD_BITNESS.exe"`
and DEPLOY_URL being set as such:
`DEPLOY_URL="https://www.mudlet.org/wp-content/files/Mudlet-${VERSION}-windows-$BUILD_BITNESS-installer.exe"`
scp will upload the $installerExePath file, so does DEPLOY_URL need to match this?

Reply:
Old code from appveyor from when we could upload directly 😅 we can't do it this way now with the GitHub switch. Need to think on a new solution. 

Maybe we make the queue mechanism handle release builds as well and just place them in the right location on the server?

Reply:
Does the file need to be in the right place before running the dblsqd release/push? I could just let it upload using the artifact queue.. Then a manual move or script you run on www.mudlet.org to do the wp-downloadmanager thing?

Reply:
> Does the file need to be in the right place before running the dblsqd release/push? 

Yeah, as that is the location that will be used when updating from one version to the next.

Reply:
> upload using the artifact queue

Perhaps we adjust the queue, add an additional parameter to say that this upload is a release build and it should go into a different location?